### DISCUSSION TITLE
VTOL Fixed-wing model

### DISCUSSION THREAD

**Discussion by jonyMarino** [JSBSim-Team/jsbsim#871] [Q&A]
Hi,

I want to use a fixed-wing model with VTOL capabilities (through more vertical rotors, not a tilt-rotor). Does anyone know of one that can be shared?

**Comment by jonyMarino** [comment]
I am trying to create a model by myself. I am using the base c310 model and added four pointing-up rotors.
[This](https://github.com/jonyMarino/jsbsim/tree/c310_vtol) is the branch I'm working on
[This](https://github.com/jonyMarino/jsbsim/blob/c310_vtol/aircraft/c310/c310.xml) is the link to the modified c310 model
And [this](https://github.com/jonyMarino/jsbsim/blob/c310_vtol/scripts/c3101.xml) is the test script.
The issue is that the altitude stays the same over time, even increasing the hp.
What can be happening?

**Comment by seanmcleod** [comment]
Are you sure a pitch attitude of -90 is correct for the 4 vertical motors as opposed to +90?

https://github.com/jonyMarino/jsbsim/blob/096bfb5902f024222919cc37b780ed5168bdaadd/aircraft/c310/c310.xml#L320-L324

**Comment by seanmcleod** [comment]
Plus, log the propulsion forces, which will immediately tell you first of all whether the propulsion magnitude is greater than the weight of the aircraft, and also confirm the orientation of the propulsion forces.

https://github.com/JSBSim-Team/jsbsim/blob/10c6e52e74452c31cd8dc2b857e2abbfe7b46daa/src/models/FGPropulsion.cpp#L819-L821

**Comment by jonyMarino** [comment]
@seanmcleod Thanks for getting back to me so quickly.
I have changed the rotors' pitch attitude to 90, and I print the aircraft's attitude, and what I see it´s happening is that the plane rotates, which makes it crash into the ground instead of going up without changing the orientation. 
If you want to run it by yourself, I've added these [script](https://github.com/jonyMarino/jsbsim/blob/c310_vtol/scripts/c3101b.xml), [engine](https://github.com/jonyMarino/jsbsim/blob/c310_vtol/engine/engIO470Db.xml) and [aircraft](https://github.com/jonyMarino/jsbsim/blob/c310_vtol/aircraft/c310/c310b.xml) files

**Reply by eferreirafilho** [reply]
Hi @jonyMarino , have you ever got the VTOL working nicely? Thanks!

**Reply by jonyMarino** [reply]
Hi @eferreirafilho . Unfortunately no. I haven't invested so much also, but I'm really interested in a model for vtol.

**Comment by seanmcleod** [comment]
Given that the aircraft is sitting on the ground and trying to take-off, during this initial take-off period there are no aerodynamic forces in play yet given the extremely low dynamic pressure.

So any movement and rotation is going to come down to the forces and moments from your 6 engines. So what did you see when you logged the propulsion forces I mentioned above?

Also take a look at the propulsion moments.

https://github.com/JSBSim-Team/jsbsim/blob/10c6e52e74452c31cd8dc2b857e2abbfe7b46daa/src/models/FGPropulsion.cpp#L822-L824

And lastly log the thrust of all 6 engines individually. Given that and the cg and moments of inertia of the aircraft you should be able to figure out what is going wrong.

**Comment by seanmcleod** [comment]
In addition to double-checking the forces from the engines and their location relative to the cg, and remember the cg isn't the same as the cg as specified in the FDM, since that is for the empty weight. Look at the mass report that is printed when the model is loaded to figure out the current cg at take-off with fuel and any other masses specified. Also take into account `sense` and `p-factor` when using a propellor. 

https://github.com/JSBSim-Team/jsbsim/blob/39c204a30d36edab7c3bd666abfd0640f23fc521/src/models/propulsion/FGPropeller.cpp#L258-L273

For example take a look at quadcopter example in the JSBSim repo.

https://github.com/JSBSim-Team/jsbsim/blob/master/aircraft/F450/Propulsion.xml
